conda_lock/pypi_solver.py

```diff
@@ -377,3 +381,10 @@ def _prepare_repositories_pool(allow_pypi_requests: bool) -> Pool:
     if allow_pypi_requests:
         repos.append(PyPiRepository())
     return Pool(repositories=[*repos])
+
+
+def _strip_auth(url: str) -> str:
+    """Strip HTTP Basic authentication from a URL."""
+    parts = urlsplit(url, allow_fragments=True)
+    netloc = parts.netloc.replace(f"{parts.username}:{parts.password}@", "")
```

Comment:
I'm worried about this somehow being too clever and missing something, especially since URLs sometimes get encoded. Would this work?

```suggestion
    # Remove everything before and including the last '@' character in the part
    # between 'scheme://' and the subsequent '/'.
    netloc = parts.netloc.split("@")[-1]
```

Reply:
That seems sensible 👍